### PR TITLE
kubevirt: init at 0.41.0

### PR DESCRIPTION
**Pull request by nshalman**
###### Motivation for this change

I want to be able to use this software. I imagine others might as well.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/kubevirt/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  vendorSha256 = null;
+
+  subPackages = [ "cmd/virtctl" ];
+
+  meta = with lib; {
+    description = "A virtual machine management add-on for Kubernetes";
+    homepage = "https://github.com/kubevirt/kubevirt";
+    maintainers = with maintainers; [ nshalman ];
+    license = lib.licenses.asl20;
```
Comment (fabaff) [inline pkgs/development/tools/kubevirt/default.nix:22]: `meta` has `with lib;`

```suggestion
    license = licenses.asl20;
```

File: pkgs/development/tools/kubevirt/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "kubevirt";
+  version = "0.41.0";
+
+  src = fetchFromGitHub {
+    owner = "kubevirt";
+    repo = "kubevirt";
```
Comment (fabaff) [inline pkgs/development/tools/kubevirt/default.nix:9]: Opinionated change:

```suggestion
    owner = pname;
    repo = pname;
```

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 125561` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kubevirt</li>
  </ul>
</details>

**Review by fabaff — COMMENTED** [review]
It would be nice have the version in the output. Could you please check if it possible to adjust `ldflags` (usually the Makefile contains hints)? 
 
```bash
...ew/pr-125561]$ ./results/kubevirt/bin/virtctl version
Client Version: version.Info{GitVersion:"v0.0.0-master+$Format:%h$", GitCommit:"$Format:%H$", GitTreeState:"", BuildDate:"1970-01-01T00:00:00Z", GoVersion:"go1.16.4", Compiler:"gc", Platform:"linux/amd64"}
dial tcp [::1]:8080: connect: connection refused
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -13533,6 +13533,8 @@ in
 
   kubicorn = callPackage ../development/tools/kubicorn {  };
 
+  kubevirt = callPackage ../development/tools/kubevirt {  };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:13536]:
```suggestion
  kubevirt = callPackage ../development/tools/kubevirt { };
```

File: pkgs/development/tools/kubevirt/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  vendorSha256 = null;
+
+  subPackages = [ "cmd/virtctl" ];
+
+  meta = with lib; {
+    description = "A virtual machine management add-on for Kubernetes";
+    homepage = "https://github.com/kubevirt/kubevirt";
+    maintainers = with maintainers; [ nshalman ];
+    license = lib.licenses.asl20;
+    platforms = platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/kubevirt/default.nix:23]:
```suggestion
```

already set by buildGoModule

**Comment by nshalman** [comment]
> It would be nice have the version in the output. Could you please check if it possible to adjust `ldflags` (usually the Makefile contains hints)?

They appear to use some sort of complex Dockerized Bazel build: https://github.com/kubevirt/kubevirt/blob/master/Makefile

We could also download and package their official binaries because it's written in Go...

I'll see if I can figure out where the version information is supposed to come from.

Thanks everyone for the reviews!

**Comment by nshalman** [comment]
Looks like this might be what I need to emulate:
https://github.com/kubevirt/kubevirt/blob/master/hack/build-go.sh#L127

**Comment by SuperSandro2000** [comment]
> We could also download and package their official binaries because it's written in Go...

Lets avoid that.

It is manually set or here https://github.com/kubevirt/kubevirt/blob/master/hack/common.sh#L58

**Comment by nshalman** [comment]
The kubevirt build system is proving too scary for me. If someone else wants to pick this up and carry it forward, feel free, but make sure to remove me from the maintainers list. 😄 

I don't think I will be doing this any time soon.

**Comment by haslersn** [comment]
@nshalman see #171734

**Pull request closed without merging**